File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.py
"""
문자열 myString이 주어집니다. myString을 문자 "x"를 기준으로 나눴을 때 나눠진 문자열 각각의 길이를 순서대로 저장한 배열을 return 하는 solution 함수를 완성해 주세요.
"""

def solution(myString):
    answer = []
    cnt = 0
    for str in myString:
        if str == "x":
            answer.append(cnt)
            cnt = 0
        else:
            cnt += 1
            
    answer.append(cnt)
    return answer

File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.java
import java.util.*;

class Solution {
    public int[] solution(String myString) {
        List<Integer> answer = new ArrayList<Integer>();
        
        int cnt = 0;
        for (char c : myString.toCharArray()) {
            if (c == 'x') {
                answer.add(cnt);
                cnt = 0;
            } else {
                cnt++;
            }
        }
        
        answer.add(cnt);
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}